content/varnish-configs.md

```diff
@@ -0,0 +1,192 @@
+|------ |------------------------ |------------ |-----------  |
+| east  | 2 Backends              | 2 Backends  | 1 Backend   |
+| west  | 3 Backends, (1 Canary)  | 2 Backends  | 1 Backend   |
+| eu    | 1 Backend               | 1 Backend   | N/A         |
+| syd   | N/A                     | 1 Backend   | N/A         |
+
+N.B. The registry search Service is not based on location and randomly
+chooses east or west, regardless of requesting IP.
+
+## Caching Responses
+
+In order to improve performance, npm caches the response to many requests.
+
+When a request is received, Fastly will check if the Backend response is
+something that should be cached:
+
+- Responses to requests for package tarballs, metadata, search data, etc., 
+are cached for specific durations depending on the type of data they 
+contain. 
+- Requests that fail and scoped packages are not cached. 
+
+More specificially:
+
+- If the Backend response (`beresp`) is Not Found (`404`), Internal Server
+  Error (`500`), or Service Unavailable (`503`) and a restart hasn't been
+  tried, try a restart
+
+  - If that restart doesn't work, tell Varnish to not use any cache and not
+    to cache. More specifically:
+
+    - set the header's `Cache-Control` property to `max-age=0`
+      (NOTE: this affects both Varnish and the browser)
+    - set the time-to-live (`ttl`) to 0
+    - not perform a lookup to see if an object is in cache
+    - not cache the response from the origin
+
+- If the request URL contains an `@`, indicating it is a scoped package,
+  do not cache, exactly as above. Any caching would potentially leak 
+  information. (Scoped packages are potentially private.)
+
+- Otherwise:
+  
+  - If the request is for a tarball, cache for 6 hours, or
+  - If the request is a search request, cache for 1 hour, or
+  - Cache for 5 minutes
+
+  then deliver the cached response.
+
+## Putting It All Together
+
+Now that we have defined an ACL, Directors, and a caching strategy, we can
+synthesize a coherent request handling process. This is defined in `vcl_recv`.
+
+  1. Receive all requests to `regsitry.npmjs.org`
+  2. Reject all IPs listed in the the ACL
+  3. Determine if the request is for a tarball (package data) or something else
+
+From here, we handle the request in one of two ways:
+
+### Tarball Requests
+
+Tarball requests are requests for package data. These requests hit the 
+package servers. These are only `GET` requests.
+
+If a tarball request is received:
+    
+1. On the first attempt, try to download the tarball from EC2/nginx. To
+  do so we'll need to update the request URL:
+
+  - **Scoped Packages**
+
+    ```
+    /@bcoe/thumbd/-/thumbd-2.16.2.tgz ---> /npm/public/registry/@/@foo/_attachments/foo-0.2.2.tgz
+    ```
+
+  - **All Other Packages**
+
+    ```
+    /foo/-/foo-0.2.2.tgz ---> /npm/public/registry/f/foo/_attachments/foo-0.2.2.tgz
+    ```
+
+2. After updating the URL, send the request to the appropriate `frontdoor` server
+   location:
+
```

Comment:
super agree. my first attempt at docs is always **understand wtf is happening**. i think you are right about this and i'll step up a level of abstraction 😅